File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.cpp
class Solution {
public:
    bool hasIncreasingSubarrays(vector<int>& nums, int k) {
        if(k == 1)
        {
            return true;
        }
        unordered_set<int> incSubs;
        int l = 0;
        for(int i = 1; i < nums.size(); i++)
        {
            if(nums[i] > nums[i - 1])
            {
                if(i - l + 1 == k)
                {
                    if(incSubs.find(l - k) != incSubs.end())
                    {
                        return true;
                    }
                    incSubs.insert(l);
                    l++;
                }
            }
            else
            {
                l = i;
            }
        }
        return false;
    }
};

File: 3612-adjacent-increasing-subarrays-detection-i/3612-adjacent-increasing-subarrays-detection-i.py
class Solution:
    def hasIncreasingSubarrays(self, nums: List[int], k: int) -> bool:
        if k == 1:
            return True
        incSubs = set()
        l = 0
        for r in range(1, len(nums)):
            if nums[r] > nums[r - 1]:
                if r - l + 1 == k:
                    if l - k in incSubs:
                        return True
                    incSubs.add(l)
                    l += 1
            else:
                l = r
        return False